### DISCUSSION TITLE
Return Data if relation does exist. If not, return blueprint.

### DISCUSSION THREAD

**Discussion by ahoiroman** [spatie/laravel-data#295] [Q&A]
Hello,

I'd like to solve an issue. In my controller I am calling:

```php
CompanyData::from($company->loadMissing('location'));
```

If a new *Company* does not have a location yet, the result is always `NULL`, although the relation is loaded. 

This can be explained by looking at the *RelationalLazy*. class:

```php
public function resolve(): mixed
{
    return $this->model->{$this->relation} !== null ? ($this->value)() : null;
}
```

Okay - so now I created a little workaround:

```php
location     : Lazy::when(fn() => $model->relationLoaded('location'),
                   fn() => $model->location()->exists() ? LocationData::optional($model->location)
                       : LocationData::empty()),
```

As you can see here, I am returning the `LocationData` if the location does exist - and if not I am returning a blueprint, as my Vue-frontend wants to work with `company.location`.

My question: Is there a more elegant way to solve this issue?

**Comment by Nir-An** [comment]
Hi @ahoiroman  :+1: 

if I understand correctly 
 [Relational Lazy properties](https://spatie.be/docs/laravel-data/v2/as-a-resource/lazy-properties#content-relational-lazy-properties) could be helpful 


```php

Lazy::whenLoaded('location', $company, fn() => $company->location);

```
for default data issue you could use [eloquent-relationships#default-models](https://laravel.com/docs/9.x/eloquent-relationships#default-models)  

According to Laravel Docs : 
>often referred to as the [Null Object pattern](https://en.wikipedia.org/wiki/Null_Object_pattern) and can help remove conditional checks in your code
